misk/src/main/kotlin/misk/tracing/interceptors/TracingMethodInterceptor.kt

```diff
@@ -0,0 +1,34 @@
+package misk.tracing.interceptors
+
+import io.opentracing.Scope
+import io.opentracing.Tracer
+import io.opentracing.tag.Tags
+import misk.logging.getLogger
+import org.aopalliance.intercept.MethodInterceptor
+import org.aopalliance.intercept.MethodInvocation
+import javax.inject.Inject
+import javax.inject.Provider
+
+private val logger = getLogger<TracingMethodInterceptor>()
+
+class TracingMethodInterceptor @Inject internal constructor(
```

Comment:
nit: should we have a single `TracingInterceptor` that implements both `MethodInterceptor` and `Interceptor`? might be able to DRY up some of the boilerplate, but not sure

Reply:
Yeah, will give that a go and see how it looks.